src/pages/index.js

```diff
@@ -1,127 +1,248 @@
+          .finally(() => {
+            setSubmitButtonText(deleteCardModalBtn, "Yes");
+          });
+      });
+      deleteCardPopup.open();
+    },
+    handleLikeClick: () => {
+      if (cardElement.isLiked()) {
+        api
+          .likeCountRemove(card._id)
+          .then((card) => {
+            cardElement.setLikes(card.likes);
+          })
+          .catch((err) => {
+            console.error(err.status);
+          });
+      } else {
+        api.likeCountAdd(card._id).then((card) => {
+          cardElement.setLikes(card.likes);
+        });
+      }
+    },
+    userId: userId,
+  });
 
+  return cardElement.getView();
+}
 
-//        Modal Image       ===============================================
+//   Card Image Modal  =================================================
 
 const cardImagePopup = new PopupWithImage(selectors.cardImageModal);
 
 cardImagePopup.setEventListeners();
 
-// Card Section ================================================================
+// Delete Card Modal  ====================================================
+
+const deleteCardPopup = new PopupWithConfirmation(selectors.deleteCardModal);
+deleteCardPopup.setEventListeners();
+
+// Add Card  ===============================================================
+
+const addCardPopup = new PopupWithForm(selectors.addCardModal, (data) => {
+  setSubmitButtonText(addCardSubmitBtn, "Creating...");
+
+  api
+    .addCard(data)
+    .then((data) => {
+      const newCard = renderCard(data, userId);
+      sectionInstance.addItem(newCard);
+    })
+    .catch((err) => {
+      console.error(err.status);
+    })
+    .then(() => {
+      addCardPopup.close();
+      formValidators["add-card-form"].disableButton()
+    })
+    .finally(() => {
+      setSubmitButtonText(addCardSubmitBtn, "Create");
+    });
+});
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
+addCardBtn.addEventListener("click", () => {
+  addCardPopup.open();
+});
 
-      const card = renderCard(data)
+addCardPopup.setEventListeners();
 
-      cardSection.addItem(card);
-    },
-  },
+// Profile  ================================================================
 
-  selectors.cardList
+const userInfo = new UserInfo(
+  selectors.profileTitle,
+  selectors.profileDescription,
+  selectors.profileImage
 );
 
-cardSection.renderItems(initialCards);
-
-// Add Card Form ===============================================================
-
-
-const addCardFormPopup = new PopupWithForm(selectors.addCardModal, (data) => {
-  const newCard = renderCard(data)
+let sectionInstance;
+let userId;
+
+const editProfilePopup = new PopupWithForm(selectors.profileModal, (data) => {
+  setSubmitButtonText(profileModalSubmitBtn, "Saving...");
+  console.log(data);
+  api
+    .editProfile(data)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      editProfilePopup.close();
+      formValidators["profile-edit-form"].disableButton();
```

Comment:
`.disableButton()` should be called when you **open** modals because when you close them you reset the form according to the task